src/sync/atomic/mod.rs

```diff
@@ -159,6 +175,8 @@ impl<T: Copy + Eq> Atomic<T> {
         maybe_warn_about_ordering(order);
 
         thread::switch();
+        self.inhale_clock();
+        self.exhale_clock();
```

Comment:
Why do these happen in the opposite order to in `fetch_update`?

Reply:
Nice catch.  It should be the other way around.  I also realized fetch_update was wrong -- it was not handling the error case properly. :-(

Fixed both and added comments to explain.

Reply:
Also added a new test `clock_fetch_update` to reveal the fetch_update bug.  For the other one, I think the way I had it wasn't a bug, but reordering the steps is more consistent with translating the swap to load/store.